querygen/querygen.go

```diff
@@ -57,3 +57,33 @@ func GenBuildPipelineRunStartedQuery(index string) string {
 	)
 	return q
 }
+
+// GenClairScanCompletedQuery returns a Splunk query for generating Segment events
+// when the clair-scan task completes.
+func GenClairScanCompletedQuery(index string) string {
+	q, _ := UserJourneyQueryGen(
+		index,
+		`verb=update `+
+			`"responseStatus.code"=200 `+
+			`"objectRef.apiGroup"="tekton.dev" `+
+			`"objectRef.resource"="taskruns" `+
+			`"objectRef.subresource"="status" `+
+			`"requestObject.metadata.labels.tekton.dev/pipelineTask"="clair-scan" `+
+			`"responseObject.status.completionTime"="*" `+
+			`| spath responseObject.status.conditions{} `+
+			`| mvexpand responseObject.status.conditions{} `+
+			`| search responseObject.status.conditions{}.type="Succeeded" `+
+			`responseObject.status.conditions{}.reason="Succeeded" `+
+			`responseObject.status.conditions{}.status="True" `+
+			`| spath responseObject.status.taskResults{} `+
+			`| mvexpand responseObject.status.taskResults{} `+
+			`| search responseObject.status.taskResults{}.name="CLAIR_SCAN_RESULT" `+
+			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.critical output=clair_scan_result.vulnerabilities.critical `+
+			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.high output=clair_scan_result.vulnerabilities.high `+
+			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.medium output=clair_scan_result.vulnerabilities.medium `+
+			`| spath input=responseObject.status.taskResults{}.value path=vulnerabilities.low output=clair_scan_result.vulnerabilities.low `+
+			`| eval event="Clair scan TaskRun completed"`,
+		[]string{"namespace", "application", "component", "vulnerabilities_critical", "vulnerabilities_high", "vulnerabilities_medium", "vulnerabilities_low"},
+	)
```

Comment:
This result is not good - this is a hugs messy and unreadable string.

could we generate some of those fields via our existing FiedlSet mechanism?

I understand we may have a need to run search expressions after having generated some field values, we may need to expand FieldSet a bit to support this.

Mixing field value generation in what is supposed to only be a predicate string, while I thought to be tolerable at first, is not longer looking to me like it would be sustainable approach.

Specifically WRT the `spath` statements, it seems `eval` has an [spath function][1] that could be used in conjunction with our existing `FieldSet` code (If this become commonly used, I would, however, add a field to the FieldSet struct to simplify this case.

[1]: https://docs.splunk.com/Documentation/Splunk/9.1.0/SearchReference/TextFunctions#spath.28.26lt.3Bvalue.26gt.3B.2C.26lt.3Bpath.26gt.3B.29